pkg/schema/tracker_test.go

```diff
@@ -34,10 +34,51 @@ var _ = Suite(&trackerSuite{})
 
 type trackerSuite struct{}
 
+func (s *trackerSuite) TestSessionCfg(c *C) {
+	log.SetLevel(zapcore.ErrorLevel)
+
+	sessionCfg := map[string]string{"sql_mode": "HaHa"}
+	tracker, err := schema.NewTracker(sessionCfg)
+	c.Assert(err, NotNil)
+
+	tracker, err = schema.NewTracker(nil)
+	ctx := context.Background()
+	err = tracker.Exec(ctx, "", "create database testdb;")
+	c.Assert(err, IsNil)
+
+	// Now create the table with ZERO_DATE
+	err = tracker.Exec(ctx, "testdb", "create table foo (a varchar(255) primary key, b DATETIME NOT NULL DEFAULT '0000-00-00 00:00:00')")
+	c.Assert(err, NotNil)
+
+	// set session config
+	sessionCfg = map[string]string{"sql_mode": "NO_ZERO_DATE,NO_ZERO_IN_DATE,ANSI_QUOTES"}
+	tracker, err = schema.NewTracker(sessionCfg)
+	c.Assert(err, IsNil)
+
+	err = tracker.Exec(ctx, "", "create database testdb;")
+	c.Assert(err, IsNil)
+
+	// Now create the table with ANSI_QUOTES and ZERO_DATE
+	err = tracker.Exec(ctx, "testdb", "create table \"foo\" (a varchar(255) primary key, b DATETIME NOT NULL DEFAULT '0000-00-00 00:00:00')")
+	c.Assert(err, IsNil)
```

Comment:
should it return error because `b` is ZERO_DATE and SQL mode is `NO_ZERO_DATE`?

update: seems `STRICT_TRANS_TABLES` should also be set, to prevent using ZERO_DATE value. 

https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_no_zero_date